### PR TITLE
Sync upstream 20191106

### PR DESCRIPTION
**Pull request by vext01**
Sync with upstream.

We haven't found a good way of doing these that allows them to be reviewable.

Normally I resolve merge conflicts as best I can in the main merge commit, then add extra fixes needed in follow up commits. I recommend only reviewing the latter, as the former will make your head spin.

These all need to be merged into one commit so that `git bisect` can work.

### CODE REVIEW

File: src/librustc_mir/transform/add_yk_swt_calls.rs [hunk 1]
```diff
@@ -77,7 +77,7 @@ impl MirPass<'_> for AddYkSWTCalls {
             let ret_val = LocalDecl::new_temp(unit_ty, DUMMY_SP);
             let ret_place = Place {
                 base: PlaceBase::Local(Local::new(num_orig_local_decls + new_local_decls.len())),
-                projection: Box::new([]),
+                projection: List::empty(), //Box::new([]),
```
Comment (ptersilie) [inline src/librustc_mir/transform/add_yk_swt_calls.rs]: comment

**Comment by vext01** [comment]
Fixed.

**Comment by ptersilie** [comment]
Please squash!

**Comment by vext01** [comment]
splat.

**Pull request merged**